.github/workflows/test.yml

```diff
@@ -56,7 +56,7 @@ jobs:
       - name: Run tests
         run: |
           export C_INCLUDE_PATH=$pythonLocation/include/python${{ steps.python-selector.outputs.python-version }}:$C_INCLUDE_PATH
-          pytest --cov-report term-missing:skip-covered
+          pytest -n logical --cov-report term-missing:skip-covered
```

Comment:
Do we want to always activate pytest-xdist for all users when they run tests ? If yes, I would put that in `addopts` in the  `pytest.ini` config file. Also, you would need to add `pytest-xdist` and `pytest-xdist[psutil]` to the `required_plugins` list.

Reply:
Add this : `--dist worksteal`. it makes the test worker steal job from each other when they finnish early.